File: app/src/main/java/jogja/telkom/csrassistant/Tagihan/TidakSesuaiTagihanActivity.java
package jogja.telkom.csrassistant.Tagihan;

import android.content.Context;
import android.os.Bundle;

import java.util.ArrayList;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import jogja.telkom.csrassistant.Adapter.ChildInstruksi;
import jogja.telkom.csrassistant.Adapter.Instruksi;
import jogja.telkom.csrassistant.Adapter.RecAdapter;
import jogja.telkom.csrassistant.R;

public class TidakSesuaiTagihanActivity extends AppCompatActivity {
    private RecyclerView mRecyclerView;
    private Context mContext;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tagihan_tidak_sesuai);
        mContext = TidakSesuaiTagihanActivity.this;
        mRecyclerView = findViewById(R.id.recview_tak_sesuai);
        RecAdapter recyclerDataAdapter = new RecAdapter (getDummyDataToPass());
        mRecyclerView.setLayoutManager(new LinearLayoutManager(mContext));
        mRecyclerView.setAdapter(recyclerDataAdapter);
        mRecyclerView.setHasFixedSize(true);
    }

    private ArrayList<Instruksi> getDummyDataToPass() {
        ArrayList<Instruksi> dataInstruksi = new ArrayList<>();
        ArrayList<ChildInstruksi> childDataItems;
        Instruksi instruksi;
        ChildInstruksi childInstruksi;

        /*UNTUK HEAD INSTRUKSI*/
        instruksi = new Instruksi();
        instruksi.setInstruksi("3. BUKA NOSSA");
        childDataItems = new ArrayList<>();

        /*UNTUK ISI*/
        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Buka Nossa dan Lihat Halaman Tiket Pelanggan");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Pergi ke Work Logs > Isi Summary dan Details ");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Contoh Summary: Sudah Proses TAMBAH Add On XXXX dengan SC XXXXXXXX,\n" +
                " Details boleh diisi sama dengan Summary apabila tidak ada informasi tambahan ");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Save dengan menekan Icon Disket di bagian atas");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Pergi ke Tab Actual Resolution");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Pada Actual Solution, tekan icon kaca pembesar, pada Decription ketik \\\"LAYANAN\\\",\n" +
                " tekan Enter, Pilih SOLUTION_IPTC,IPTV - PAKET LAYANAN");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Pada Resolution isikan seperti pada Summary sebelumnya");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Save dengan menekan Icon Disket di bagian atas ");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Take Ownership dengan menekan icon tangan memegang kertas");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Tekan icon Route Workflow (sebelah kiri icon Take Ownership)");
        childDataItems.add(childInstruksi);

        childInstruksi = new ChildInstruksi();
        childInstruksi.setIsiInstruksi("- Pilih Resolve Incident");
        childDataItems.add(childInstruksi);

        /*UNTUK MENAMBAHKAN PADA LIST*/
        instruksi.setChildInstruksis(childDataItems);
        dataInstruksi.add(instruksi);
        /*--------------SELESAI----------------------*/

        /*-----------------*/
        return dataInstruksi;
    }
}

File: app/src/main/java/jogja/telkom/csrassistant/Tagihan/TidakSesuaiTagihanActivity.kt
package jogja.telkom.csrassistant.Tagihan

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import jogja.telkom.csrassistant.Adapter.ChildInstruksi
import jogja.telkom.csrassistant.Adapter.Instruksi
import jogja.telkom.csrassistant.Adapter.RecAdapter
import jogja.telkom.csrassistant.R
import java.util.*

class TidakSesuaiTagihanActivity : AppCompatActivity() {
    private var mRecyclerView: RecyclerView? = null
    private var mContext: Context? = null

    private val dummyDataToPass: ArrayList<Instruksi>
        get() {
            val dataInstruksi = ArrayList<Instruksi>()
            var childDataItems: ArrayList<ChildInstruksi>
            var instruksi: Instruksi
            var childInstruksi: ChildInstruksi

            instruksi = Instruksi()
            instruksi.instruksi = "CEK SISKA TOOL"
            childDataItems = ArrayList()
            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Periksa Nomor Jastel pelanggan"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Periksa apakah ada paket yang ditambahkan yang sekiranya membuat tagihan pelanggan bertambah, biasanya Add On, pertambahan kecepatan, atau fitur lain"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Apabila kesalahan tagihan akibat ketidaktahuan pelanggan terhadap pajak, langsung edukasi pelanggan dan tutup laporan"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Edukasi pelanggan (melalui telepon, WA, atau SMS) tentang hal yang membuat tagihan bertambah, dan tanyakan apakah masih berkenan untuk dicabut"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Bila berkenan cabut layanan sesuai prosedur, bila tidak tutup laporan"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Lakukan koreksi agar bulan berikutnya pelanggan tidak ditagih lagi"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Informasikan kembali jika proses penjabutan sudah selesai (melalui WA atau SMS)"
            childDataItems.add(childInstruksi)

            childDataItems.add(childInstruksi)
            instruksi.setChildInstruksis(childDataItems)
            dataInstruksi.add(instruksi)

            instruksi = Instruksi()
            instruksi.instruksi = "TUTUP LAPORAN"
            childDataItems = ArrayList()
            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Kembali ke Nossa, isi Log bahwa fitur/paket yang tidak diinginkan sudah dicabut dan pelanggan sudah diedukasi"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Pada Actual Resolution search 'Tagihan' dan pilih sesuai dengan fitur yang dicabut (Internet atau IPTV)"
            childDataItems.add(childInstruksi)

            childInstruksi = ChildInstruksi()
            childInstruksi.isiInstruksi = "Tekan tombol Take Ownership, lalu tekna tombol Route Workflow, pilih Resolve Incident"
            childDataItems.add(childInstruksi)

            childDataItems.add(childInstruksi)
            instruksi.setChildInstruksis(childDataItems)
            dataInstruksi.add(instruksi)
            return dataInstruksi
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_recview)
        mContext = this@TidakSesuaiTagihanActivity
        mRecyclerView = findViewById(R.id.layout_recview)
        val recyclerDataAdapter = RecAdapter(dummyDataToPass)
        mRecyclerView!!.layoutManager = LinearLayoutManager(mContext)
        mRecyclerView!!.adapter = recyclerDataAdapter
        mRecyclerView!!.setHasFixedSize(true)
    }
}
